.github/workflows/ci-ptf-tests.yml

```diff
@@ -0,0 +1,94 @@
+name: "P4SDE CI Build pipeline"
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+      
+concurrency:
+  # if workflow for PR or push is already running stop it, and start new one
+  group: build_p4sde_ci-${{ github.ref }}
+  cancel-in-progress: true
+
+# Envs for infrap4d and dpdk libs
+env:
+    SDE: ${{ github.workspace }}
+    P4C_DIR: ${{ github.workspace }}/p4c
+    SDE_INSTALL: ${{ github.workspace }}/install
+    IPDK_RECIPE: ${{ github.workspace }}/ipdk.recipe
+    DEPEND_INSTALL: ${{ github.workspace }}/install
+
+jobs:
+  build_p4dpdk_ubuntu:
+    runs-on: ubuntu-22.04
+    steps:
+      - name: 'ccache'
+        uses: hendrikmuhs/ccache-action@v1
+        with:
+          key: ptf-${{ runner.os }}-test
+          max-size: 1000M
+
+      - name: 'Checkout DPDK-target'
+        uses: actions/checkout@v3
+        with:
+             repository: p4lang/p4-dpdk-target
+             path: p4sde
+             submodules: 'recursive'
+
+      - name: 'Checkout ipdk-recipe'
+        uses: actions/checkout@v3
+        with:
+             repository: ipdk-io/networking-recipe
+             path: ipdk.recipe
+             submodules: 'recursive'
+
+      - name: checkout P4C
+        uses: actions/checkout@v3
+        with:
+             path: p4c
+             submodules: recursive
+
+      - name: 'Install DPDK dependencies'
+        working-directory: p4sde/tools/setup
+        run: |
+              sudo apt update -y
+              python install_dep.py
+
+      - name: 'Compile p4sde dpdk target'
+        working-directory: p4sde
+        run: |
+             mkdir ${GITHUB_WORKSPACE}/install
+             ./autogen.sh
+             ./configure --prefix=$SDE_INSTALL
+             make 
+             make install
+
+      - name: Build p4c with only the DPDK backend
+        working-directory: p4c
+        run: |
+          sudo -E tools/dpdk-ci-build.sh $P4C_DIR $IPDK_RECIPE $SDE_INSTALL $DEPEND_INSTALL
+
+      - name: 'Compile ipdk recipe'
+        run: |
+             echo "Install infrap4d dependencies"
+             cd ipdk.recipe
+             sudo apt install libatomic1 libnl-route-3-dev openssl
+             sudo pip3 install -r requirements.txt
+             cd $IPDK_RECIPE/setup
+             echo "Build infrap4d dependencies"
```

Comment:
Can you split this into two steps so we can understand how long it takes to install the infrap4d dependencies? Maybe we can cut this down.

Reply:
I believe this dependency step takes up most of the time and we can simplify it.

Reply:
Do you mean splitting dependency install and dependency build or splitting dependency build and infrap4d build?

Reply:
Line 76-83

Reply:
Could you check the updated CI to see if it is what we want for the separation?

Reply:
Yes, it shows that most of the time is spent on dependencies. If we get rid of those the tool will build fast.